pvlib/pvsystem.py

```diff
@@ -1046,28 +1047,60 @@ def temperature_model_parameters(self, value):
             array.temperature_model_parameters = value
 
     @property
-    @_unwrap_single_value
     def surface_tilt(self):
-        # TODO: don't access mount attributes directly?
-        return tuple(array.mount.surface_tilt for array in self.arrays)
+        if len(self.arrays) == 1:
+            msg = (
+                'PVSystem.surface_tilt attribute is deprecated. '
+                'Use PVSystem.arrays[0].mount.surface_tilt.'
+            )
+            warnings.warn(msg, pvlibDeprecationWarning)
+        else:
+            raise AttributeError(
+                'PVSystem.surface_tilt not supported for multi-array systems. '
+                'Use PVSystem.arrays[i].mount.surface_tilt.')
+        return self.arrays[0].mount.surface_tilt
 
     @surface_tilt.setter
     def surface_tilt(self, value):
-        # TODO: don't access mount attributes directly?
-        for array in self.arrays:
-            array.mount.surface_tilt = value
+        if len(self.arrays) == 1:
+            msg = (
+                'PVSystem.surface_tilt attribute is deprecated. '
+                'Use PVSystem.arrays[0].mount.surface_tilt.'
+            )
+            warnings.warn(msg, pvlibDeprecationWarning)
+        else:
+            raise AttributeError(
+                'PVSystem.surface_tilt not supported for multi-array systems. '
+                'Use PVSystem.arrays[i].mount.surface_tilt.')
+        self.arrays[0].mount.surface_tilt = value
 
     @property
-    @_unwrap_single_value
     def surface_azimuth(self):
```

Comment:
Are there any other "pass-trough" type properties that should be made consistent with this approach?

Reply:
I'd apply this approach to all of the pass-through properties, but I'm open to other opinions.

Reply:
here's a bug related to pass-through properties: https://github.com/pvlib/pvlib-python/issues/1192

Reply:
Just to make sure I understand this idea, given my lack of experience with the new PVSystem/Array ecosystem -- is the suggestion to apply a decorator like the one below to these properties: https://github.com/pvlib/pvlib-python/blob/master/pvlib/pvsystem.py#L1020-L1089

```python
def _check_deprecated_passthrough(func):
    @functools.wraps(func)
    def wrapper(self, *args, **kwargs):
        attr_name = func.__name__
        if len(self.arrays) == 1:
            msg = (
                f'PVSystem.{attr_name} attribute is deprecated. '
                f'Use PVSystem.arrays[i].{attr_name}.'
            )
            warnings.warn(msg, pvlibDeprecationWarning)
        else:
            raise AttributeError(
                f'PVSystem.{attr_name} not supported for multi-array '
                f'systems. Use PVSystem.arrays[i].{attr_name}.')

        return func(self, *args, **kwargs)

    return wrapper
```

Reply:
Yes. Better to make users do a little more work than to try to support a confusing API.

Reply:
#1196